syntax="proto3";

package tasklist;

service TaskService {
    rpc CreateTask (Task) returns (OperationResponse) {}
    rpc GetTasks (ChatId) returns (stream TaskResponse) {}
    rpc UpdateTask (TaskUpdate) returns (OperationResponse) {}
    rpc CompleteTask (TaskIdent) returns (OperationResponse) {}
    rpc RemoveTask (TaskIdent) returns (OperationResponse) {}
}

message ChatId {
    string id = 1;
}

message Task{
    ChatId id = 1;
    string text = 2;
}

message TaskUpdate{
    ChatId id = 1;
    int32 num = 2;
    string text = 3;
}

message TaskIdent{
    ChatId id = 1;
    int32 num = 2;
}

message OperationResponse{
    string text = 1;
}

message TaskResponse{
    int32 num = 1;
    string text = 2;
    bool is_completed = 3;
}